packages/client/hmi-client/src/utils/data-util.ts

```diff
@@ -12,81 +12,34 @@ import { Dataset, FACET_FIELDS as DATASET_FACET_FIELDS } from '@/types/Dataset';
-// 		if (ASSET_FACET_FIELDS.includes(filterField)) {
-// 			const filterValues = clause.values.map((v) => v.toString()); // array of values to filter upon
-// 			const isNot = !clause.isNot; // is the filter reversed?
-// 			const filteredModels = results.filter(
-// 				(asset) =>
-// 					// direct query against Model fields
-// 					filterValues.includes(asset[filterField as keyof Model].toString()) === isNot
-// 			);
-// 			// use splice to filter in place
-// 			results.splice(0, results.length, ...filteredModels);
-// 		}
-// 	});
-// };
-
-// Apply filter to data in place //////////// refactor this
-export const applyFacetFiltersToModels = (modelResults: Model[], filters: Filters) => {
-	if (isEmpty(filters) || isEmpty(modelResults)) {
+export const applyFacetFilters = <T extends Model | Dataset>(
+	results: T[],
+	filters: Filters,
+	resourceType: ResourceType
+) => {
+	if (isEmpty(filters) || isEmpty(results)) {
 		return;
 	}
+
 	const { clauses } = filters;
-	clauses.forEach((clause: any) => {
+	const ASSET_FACET_FIELDS: string[] =
+		resourceType === ResourceType.MODEL ? MODEL_FACET_FIELDS : DATASET_FACET_FIELDS;
+
+	clauses.forEach((clause) => {
 		const filterField: string = clause.field; // the field to filter on
 		// "filters" may include fields that belong to different types of artifacts
 		//  thus make sure to only filter models using Model fields
-		if (MODEL_FACET_FIELDS.includes(filterField)) {
+		if (ASSET_FACET_FIELDS.includes(filterField)) {
 			const filterValues = clause.values.map((v) => v.toString()); // array of values to filter upon
 			const isNot = !clause.isNot; // is the filter reversed?
-			const filteredModels = modelResults.filter(
-				(model) =>
-					// direct query against Model fields
-					filterValues.includes(model[filterField as keyof Model].toString()) === isNot
-			);
-			// use splice to filter in place
-			modelResults.splice(0, modelResults.length, ...filteredModels);
-		}
-	});
-};
 
-export const applyFacetFiltersToDatasets = (datasetResults: Dataset[], filters: Filters) => {
-	if (isEmpty(filters) || isEmpty(datasetResults)) {
-		return;
-	}
-	const { clauses } = filters;
-	clauses.forEach((clause: any) => {
-		const filterField: string = clause.field; // the field to filter on
-		// "filters" may include fields that belong to different types of artifacts
-		//  thus make sure to only filter datasets using Dataset fields
-		if (DATASET_FACET_FIELDS.includes(filterField)) {
-			const filterValues = clause.values.map((v) => v.toString()); // array of values to filter upon
-			const isNot = !clause.isNot; // is the filter reversed?
-			const filteredDatasets = datasetResults.filter(
-				(dataset) =>
-					// direct query against Dataset fields
-					filterValues.includes(dataset[filterField as keyof Dataset].toString()) === isNot
+			results.splice(
+				0,
+				results.length,
+				...results.filter(
+					(asset) => filterValues.includes(`${asset[filterField as keyof T]}`) === isNot
```

Comment:
Why did you remove the call to "toString()"?

Reply:
I ran into this issue so that was my attempt to work around it.
![image](https://user-images.githubusercontent.com/28237258/210811219-e0887381-9822-4ead-90e0-5679b080dfbc.png)

Is there a better way?

Reply:
I would consider casting to any first

Reply:
Changed it let me know if it's good